packages/graphql/src/utils/get-token.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IncomingMessage } from "http";
+import Debug from "debug";
+import { Context } from "../types";
+import { DEBUG_AUTH } from "../constants";
+
+const debug = Debug(DEBUG_AUTH);
+
+export function getToken(context: Context): string | undefined {
+    const req = context instanceof IncomingMessage ? context : context.req || context.request;
+    let token: string | undefined;
+
+    if (!req) {
+        debug("Could not get .req or .request from context");
+
+        return token;
+    }
+
+    if (!req.headers && !req.cookies) {
+        debug(".headers or .cookies not found on req");
+
+        return token;
+    }
+
+    const authorization = (req.headers.authorization || req.headers.Authorization || req.cookies?.token) as string;
```

Comment:
**nitpick**: Sorry for being a pain in the arse, but I think we need to be strict with our typings as much as possible
```suggestion
    const authorization = (req.headers.authorization || req.headers.Authorization || req.cookies?.token) as string | undefined;
```